gateway/blocks_gateway.go

```diff
@@ -153,15 +153,25 @@ func (api *BlocksGateway) Get(ctx context.Context, path ImmutablePath) (ContentP
 		return md, nil, err
 	}
 
-	if _, ok := f.(files.Directory); ok {
+	if d, ok := f.(files.Directory); ok {
 		dir, err := uio.NewDirectoryFromNode(api.dagService, nd)
 		if err != nil {
 			return md, nil, err
 		}
-		return md, &wrappedDirectory{node: f, dir: dir.EnumLinksAsync(ctx)}, nil
+		sz, err := d.Size()
+		if err != nil {
+			return ContentPathMetadata{}, nil, fmt.Errorf("could not get cumulative directory graph size: %w", err)
```

Comment:
nit: we always used DAG

```suggestion
			return ContentPathMetadata{}, nil, fmt.Errorf("could not get cumulative directory DAG size: %w", err)
```